websites/C/Coub/presence.ts

```diff
@@ -312,26 +301,28 @@ const pages: PageContext[] = [
 	],
 	presence = new Presence({
 		clientId: "818598086984728576",
-	});
-
+	}),
+	presenceImageKeys = {
+		PLAY: "playx1024",
+		PAUSE: "pausex1024",
+	};
+function getStrings(newLang?: string) {
+	return presence.getStrings(
+		{
+			browsing: "presence.activity.browsing",
+			watching: "presence.playback.playing",
+			watchVideo: "general.buttonWatchVideo",
+			viewProfile: "general.buttonViewProfile",
+		},
+		newLang
+	);
+}
 let currentLang: string, localizedStrings: { [key: string]: string };
-const IMAGES = {
-	PLAY: "playx1024",
-	PAUSE: "pausex1024",
-};
 presence.on("UpdateData", async () => {
-	const newLang = await presence.getSetting<string>("lang");
-	if (newLang !== currentLang) {
+	const newLang = await presence.getSetting<string>("lang").catch(() => null);
```

Comment:
why null? We usually catch this to "en"

Reply:
Presence.getStrings accepts null/undefined, thought it would resolve to "en" if null/undefined

Reply:
That’s not how JS works, when you use .catch and the promise rejects it will resolve to the value the catch function returns